src/frontend/components/home/FeedCard.jsx

```diff
@@ -17,41 +20,96 @@ const FeedCard = ({ pInfo}) => {
         else {
             setLikedDisplay(() => false)
         }
+      
+
     },[pInfo])
-    const likeHandler = async() => {
-        if(!likedDisplay){
+    useEffect(() => {
+        const isPostBookmarked = bookmarksData.find((bookmark) => bookmark._id === pInfo._id)
+        if(isPostBookmarked){
+            setBookmarkDisplay(() => true)
+        }
+        else {
+            setBookmarkDisplay(() => false)
+        }
+    },[bookmarksData])
+    useEffect(() => {
+        const isFollowing = userProfileData.following.find((user) => user.firstName + user.lastName === postUploader.firstName + postUploader.lastName)
+        if(isFollowing){
+            setFollowDisplay(() =>true)
```

Comment:
setFollowingDisplay(true) would also solve the purpose. Any reason for using arrow notation here ?

Reply:
Updater functions are more reliable in state update